core/cloudflow-operator/src/main/scala/cloudflow/operator/runner/Runner.scala

```diff
@@ -407,14 +410,39 @@ object PodsConfig {
    * Currently only dealing with secrets inside volumes
    * Not taking into acccount other type of sources
    */
+  /**
+   * Currently only dealing with secrets inside volumes
+   * Not taking into acccount other type of sources
+   *
+   * How to make this optional?
+   */
   implicit val sourceConfReader: ValueReader[Volume.Source] = ValueReader.relative { config =>
-    config.as[Volume.Secret]("secret")
+    println(config)
+    val c: Config                        = config
+    val m: java.util.Map[String, AnyRef] = c.root().unwrapped()
+    val res: Option[Volume.Source] = m.keySet().toArray().headOption.map {
+      case "secret" =>
+        config.as[Volume.Secret]("secret")
+      case "pvc" =>
+        config.as[Volume.PersistentVolumeClaimRef]("pvc")
+      case _ =>
+        logger.error(s"volume definition in config: [$config] not 'secret' nor 'pvc'. These are the only options")
+        Volume.GenericVolumeSource(config.toString)
+    }
+    res.getOrElse(Volume.GenericVolumeSource(config.toString))
   }
 
   implicit val secretsConfReader: ValueReader[Volume.Secret] = ValueReader.relative { config =>
     Volume.Secret(secretName = config.as[String]("name"))
   }
 
+  implicit val persistentVolumeClaimRefConfReader: ValueReader[Volume.PersistentVolumeClaimRef] = ValueReader.relative { config =>
+    Volume.PersistentVolumeClaimRef(
+      claimName = config.as[String]("name"),
+      readOnly = config.as[Boolean]("readOnly")
```

Comment:
readOnly should be optional (default is false). Also, other items are using hyphen, so `read-only`?